packages/block-library/src/navigation/index.php

```diff
@@ -133,8 +133,27 @@ function render_block_navigation( $content, $block ) {
 		return $content;
 	}
 
-	$attributes           = $block['attrs'];
 	$block['innerBlocks'] = gutenberg_remove_empty_navigation_links_recursive( $block['innerBlocks'] );
+	$attributes           = $block['attrs'];
+
+	// Deprecated:
+	// The rgbTextColor and rgbBackgroundColor attributes
+	// have been deprecated in favor of
+	// customTextColor and customBackgroundColor ones.
+	// Move the values from old attrs to the new ones.
+	if ( isset( $attributes['rgbTextColor'] ) ) {
+		if ( ! isset( $attributes['textColor'] ) ) {
+			$attributes['customTextColor'] = $attributes['rgbTextColor'];
+		}
+		unset( $attributes['rgbTextColor'] );
+	}
+
+	if ( isset( $attributes['rgbBackgroundColor'] ) ) {
+		if ( ! isset( $attributes['backgroundColor'] ) ) {
+			$attributes['customBackgroundColor'] = $attributes['rgbBackgroundColor'];
+		}
+		unset( $attributes['rgbBackgroundColor'] );
+	}
```

Comment:
This could be tightened up to:

```php
 	if ( isset( $attributes['rgbTextColor'] ) && empty( $attributes['textColor'] ) ) {
 		$attributes['customTextColor'] = $attributes['rgbTextColor'];
 	}
 	if ( isset( $attributes['rgbBackgroundColor'] ) && empty( $attributes['backgroundColor'] ) ) {
 		$attributes['customBackgroundColor'] = $attributes['rgbBackgroundColor'];
 	}
 	unset( $attributes['rgbTextColor'], $attributes['rgbBackgroundColor'] );
```

Reply:
`unset` with multiple attrs! 💯